catch-uncomitted.js

```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env node
+const { execFileSync } = require('child_process')
+const path = require('path')
+
+const argv = process.argv.slice(2)
+
+try {
+  execFileSync(`${path.join(__dirname, 'catch-uncomitted.sh')}`, argv, {
+    stdio: 'inherit',
+  })
+} catch (error) {
+  process.exit(error.status)
```

Comment:
```suggestion
  process.exitCode = error.status
```
 
Unless there's a specific reason to call `process.exit()`, I understand that the "graceful exit" approach is to set `process.exitCode` instead. See docs: https://nodejs.org/docs/latest-v8.x/api/process.html#process_process_exit_code